uploadNeuroDB/NeuroDB/MRI.pm

```diff
@@ -792,15 +788,39 @@ sub in_range
     return 0 unless $range_string;
     return 0 unless defined($value);
 
-    chomp($range_string);
-    if($range_string=~/^[0-9.]+$/) { ## single value element
-        return 1 if &floats_are_equal($value, $range_string, $FLOAT_EQUALS_NB_DECIMALS);
-    } else { ## range_string X-Y
-        $range_string =~ /([0-9.]+)-([0-9.]+)/;
-        return 1 if ($1 <= $value && $value <= $2) 
-            || &floats_are_equal($value, $1, $FLOAT_EQUALS_NB_DECIMALS) 
-            || &floats_are_equal($value, $2, $FLOAT_EQUALS_NB_DECIMALS);
-    }
+    # grep the min and max values of the range
+    my @range = split(/-/, $range_string);
+    my $min   = $range[0];
+    my $max   = $range[1];
+
```

Comment:
Is it possible that both `$min` and `$max` are not defined? If so, the `if`s do not cover this case. I think the tests could also be simplified like this:

```perl
return 0 if(defined $min) 
    && $value < $min && !&floats_are_equal($value, $min, $FLOAT_EQUALS_NB_DECIMALS);
return 0 if(defined $max)
    && $value > $max && !&floats_are_equal($value, $max, $FLOAT_EQUALS_NB_DECIMALS);
return 1;
```

On the other hand if you think that it's not clearer/leaner, leave it like this.

Reply:
@nicolasbrossard made the `if`s a little clearer and implemented the check for undefined `$min` and `$max` in `in_range` function as we discussed on the phone. Also simplifies the `if` statement in `identify_scan_type`